src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourcesPage.cs

```diff
@@ -83,7 +86,7 @@ public override async Task<ExternalSettingOperationResult> SetValueAsync<T>(stri
                 {
                     case MonikerPackageSources:
                         return await Task.Run(
-                            () => SavePackageSources<T>(packageSourcesList, cancellationToken),
+                            () => SavePackageSources(packageSourcesList, cancellationToken, out hasAnyHiddenPropertyChanged),
```

Comment:
Can be maybe return a tuple instead of both a value and an out param?

Reply:
It's not a pattern I've personally seen a lot. I've only really seen a combination of Try* methods where we have an out type and a return type (normally the return would be bool in this case)

https://learn.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=net-9.0, another one is https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-9.0. https://github.com/NuGet/NuGet.Client/blob/6b4db0573f22576810046c3aa1d166bc3bddf813/src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs#L42 in our repo.

With value tuples you can name things so it helps with the readability.